src/main.py

```diff
@@ -0,0 +1,119 @@
+# (c) Open Compute Project
+# (c) Google LLC
+# (c) Igor Semenov (goshik92@gmail.com)
+# 
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+
+import ocptv.output as tv
+import argparse
+import sh
+import re
+import os
+import contextlib
+from memtester_parsing import MemtesterObserver
+
+def main():
+    parser = argparse.ArgumentParser(description="OCP-compliant memtester wrapper.")
+    parser.add_argument("--mt_path", default="memtester",
+        help="Path to memtester executable")
+    parser.add_argument("--mt_args", default="",
+        help="Memtester arguments packed into a string. See `man memtester` for more detail")
+    parser.add_argument("--mt_log_filename", default=None,
+        help="Path to the file that will contain raw memtester output (both STDOUT and STDERR)")
+    args = parser.parse_args()
+
+    run = tv.TestRun(name="memtester", version="1.0")
+    dut = tv.Dut(id=sh.hostid().strip(), name=sh.hostname().strip())
+    observer = MemtesterObserver()
+    
+    # Use ExitStack() to avoid deep nesting
+    with contextlib.ExitStack() as estack:
+        estack.enter_context(run.scope(dut=dut))
+        estack.enter_context((step := run.add_step("run-memtester")).scope())
+
+        bad_addrs = step.start_measurement_series(name="bad_addrs")
+        
+        # Check if a supported version of memtester is installed in the system
+        def version_callback(version, is_known):
+            step.add_log(tv.LogSeverity.INFO, "Memtester v{} was found".format(version))
+            if (not is_known):
+                m = "This version of memtester was not tested. Expect parsing errors"
+                step.add_log(tv.LogSeverity.WARNING, m)
+        observer.callbacks.version_ready = version_callback
+        
+        # Report loop results
+        def loop_callback(loop):
+            tests = loop.failed_tests()
```

Comment:
id argue the cardinality of `loop` here should be equal to `tv.step`, but it could also be that a step is iterating thru a memory zone multiple times. opinion?

Reply:
That was my initial thought too. But then I realized that the user may request 1000 iterations, which will generate a thousand test steps. I think something unique rather than generated would better qualify for a test step. Memtester installation and version checks could be separate test steps, IMO.

Reply:
"memtester installation"? or am i reading that wrong and it's actually "memtester installation check"?

i dont think there's something wrong in having a dynamic number of steps; it doesnt have to be encoded in source code (in that the cardinality of steps is static).

maybe we dont need to do this in this PR. i'd also ask @dframe-google what he thinks

Reply:
>"memtester installation check"
This one. "check**s**" implies this.

Reply:
yeah, i was reading it wrong (X and Y checks = X check and Y check), didnt see the 's'. I'm gonna close this for now, because the PR has been up for a while, but maybe you can ping @dframe-google on the opinion here.